src/formatter/helpers/test_case_attempt_parser_spec.ts

```diff
@@ -0,0 +1,101 @@
+import { describe, it } from 'mocha'
+import { expect } from 'chai'
+import * as messages from '@cucumber/messages'
+import { parseTestCaseAttempt } from '.'
+import { getBaseSupportCodeLibrary } from '../../../test/fixtures/steps'
+import StepDefinitionSnippetBuilder from '../step_definition_snippet_builder'
+import { ParameterTypeRegistry } from '@cucumber/cucumber-expressions'
+import { parse } from '../../../test/gherkin_helpers'
+import { reindent } from 'reindent-template-literals'
+import { TestStepResult } from '@cucumber/messages'
+import { getTestCaseAttempts } from '../../../test/formatter_helpers'
+
+describe('TestCaseAttemptParser', () => {
+  describe('parseTestCaseAttempt', () => {
+    const cwd = ''
+    const supportCodeLibrary = getBaseSupportCodeLibrary()
+    const snippetSyntax = {
+      build: () => 'snippet',
+    }
+
+    const snippetBuilder = new StepDefinitionSnippetBuilder({
+      snippetSyntax,
+      parameterTypeRegistry: new ParameterTypeRegistry(),
+    })
+
+    const source = {
+      data: reindent(`
+        Feature: my feature
+          Scenario: my scenario
+            Given a passing step
+      `),
+      uri: 'a.feature',
+    }
+
+    describe('with no test step result', () => {
+      it('initialize step result with status UNKNOWN', async () => {
+        // Arrange
+        const parsed = await parse(source)
+
+        const testCase = {
+          id: '0',
+          pickleId: parsed.pickles[0].id,
+          testSteps: [
+            {
+              id: '1',
+              pickleStepId: parsed.pickles[0].steps[0].id,
+              stepDefinitionIds: [supportCodeLibrary.stepDefinitions[0].id],
+            },
+          ],
+        }
+
+        const testCaseAttempt = {
+          gherkinDocument: parsed.gherkinDocument,
+          pickle: parsed.pickles[0],
+          testCase,
+          attempt: 0,
+          willBeRetried: false,
+          stepAttachments: {},
+          stepResults: {},
+          worstTestStepResult: new TestStepResult(),
+        }
```

Comment:
This is a lot of mocking out. Thoughts on creating a testCaseAttempt in the same way as the other and then setting stepResults to an empty object?